feat: Add metrics for ACK codes.

jrconlin opened the pull request:
This adds metrics for
* `ua.command.ack.unknown_version` indicating ACK responses from the
  desktop UA that do not match known, send notifications.
* `ua.command.ack {code: ###}` recording ACK'd messages broken out by
  returned `code`

Closes: [SYNC-4456](https://mozilla-hub.atlassian.net/browse/SYNC-4456)

[SYNC-4456]: https://mozilla-hub.atlassian.net/browse/SYNC-4456?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

Commit:
feat: Add metrics for ACK codes.

This adds metrics for
* `ua.command.ack.unknown_version` indicating ACK responses from the
  desktop UA that do not match known, send notifications.
* `ua.command.ack {code: ###}` recording ACK'd messages broken out by
  returned `code`

Closes: SYNC-4456

Commit:
f clarify resolution of `timestamp`

pjenvey commented on autoconnect/autoconnect-ws/autoconnect-ws-sm/src/unidentified.rs:
```diff
@@ -115,8 +115,8 @@ impl UnidentifiedClient {
-            user.connected_at,
-            user.current_timestamp,
+            user.connected_at_ms,
+            user.current_timestamp_ms,
```
`current_timestamp` is indeed `ms` here as it's read from the database, having been written there from `increment_storage`, which wrote a `sortkey_timestamp` (`ms`) there.

pjenvey commented on autoconnect/autoconnect-ws/autoconnect-ws-sm/src/identified/mod.rs:
```diff
@@ -95,7 +95,7 @@ impl WebPushClient {
         flags: ClientFlags,
         connected_at: u64,
-        current_timestamp: Option<u64>,
+        current_timestamp_us: Option<u64>,
```
the unidentified code supplies this, it's `ms` there. The only `us` I see supplied is from a test, which is a bug in the test (and I think we probably have other similar incorrect timestamp usages in other tests)

jrconlin commented on autoconnect/autoconnect-ws/autoconnect-ws-sm/src/identified/on_server_notif.rs:
```diff
@@ -210,26 +210,29 @@ impl WebPushClient {
             // for topic messages. Instead, we'll use the explicitly set one we store in the User
             // record and copy into the WebPushClient struct.
-            topic_resp.timestamp.or(self.current_timestamp)
+            topic_resp
```
There is no bug, it was the product of mislabelling.

Commit:
f Add comments, simplify ack check

The notification version is the message_id. The message_id is an
encrypted string containing the uaid, chid, topic or timestamp, etc.
No need to check the channelId, since the version/message_id should
already have it.

Commit:
f remove labels

Commit:
Merge branch 'master' of github.com:mozilla-services/autopush-rs into feat/SYNC-4456_track_code

Commit:
Merge branch 'master' into feat/SYNC-4456_track_code

Commit:
Merge branch 'master' into feat/SYNC-4456_track_code

Commit:
f fix audit

Commit:
Merge branch 'feat/SYNC-4456_track_code' of github.com:mozilla-services/autopush-rs into feat/SYNC-4456_track_code

Commit:
Merge branch 'master' of github.com:mozilla-services/autopush-rs into feat/SYNC-4456_track_code
